test/mocks/config/mocks.h

```diff
@@ -119,5 +120,15 @@ class MockConfigProviderManager : public ConfigProviderManager {
                const Envoy::Config::ConfigProviderManager::OptionalArg& optarg));
 };
 
+class MockTypedFactory : public TypedFactory {
+public:
+  ~MockTypedFactory() = default;
```

Comment:
i think normally we try to define the ctors/dtors in the .cc file for mocks to reduce compilation times

Reply:
Done